frontend/src/component/pages/shared/Header.tsx

```diff
@@ -1,12 +1,19 @@
 import React, { Component } from 'react';
 import './Header.scss';
 
-export class Header extends Component {
+interface Props {
+  onSignOut: (event: React.MouseEvent<HTMLButtonElement>) => void;
+}
+
+export class Header extends Component<Props> {
   render() {
     return (
       <header>
         <div className={'center'}>
           <div id="logo">Short</div>
+          <button className="sign-out-btn" onClick={this.props.onSignOut}>
```

Comment:
```suggestion
          <button className="sign-out-btn" onClick={this.props.onSignOutClick}>
```

Reply:
We are delegating all the business logic to the page component. `onSignOut` contains business logic. `onSignOutClick`  solely means a click event happen, leaving the flexibility for the parent component to handle the event.